src/wovnio/wovnphp/Headers.php

```diff
@@ -62,7 +62,7 @@ public function __construct(&$env, &$store)
         }
         $this->unmaskedUrl = $this->protocol . '://' . $this->unmaskedHost . $this->unmaskedPathname;
         $this->host = $this->unmaskedHost;
-        if ($store->settings['url_pattern_name'] === 'subdomain') {
+        if (in_array($store->settings['url_pattern_name'], array('subdomain', 'custom_domain'))) {
             $intermediateHost = explode('//', $this->removeLang($this->protocol . '://' . $this->host, $this->lang()));
```

Comment:
This is not quite good enough. Since this URL will appear as an absolute URL with no path `http://french.testsite.com`, removeLang will not do anything. For custom domain we also need to append the request path in order to remove lang from the host.

Reply:
Alternatively, wouldn't changing this to `$this->removeLang($this->host, $this->lang()));` be OK? subdomain pattern does not care about the protocol anyway.

Reply:
I don't remember all the details but I too discovered that subdomain patterns specifically requires this special handling, or it wouldn't work.

Reply:
`addLangCode` method in Url.php will add protocol with subdomain pattern.
And after adding lang code, it doesn't remove protocol 😢